### PR TITLE
Ignore bridge methods when scanning for annotated methods.

### PR DESCRIPTION
**Pull request by kcooney**
This is one possible fix for #1411
The other possible fix is #1412

### CODE REVIEW

File: src/main/java/org/junit/runners/model/FrameworkMethod.java [hunk 1]
```diff
@@ -149,6 +149,11 @@ public boolean isShadowedBy(FrameworkMethod other) {
     }
 
     @Override
+    boolean isBridgeMethod() {
```
Comment (sormuras) [inline src/main/java/org/junit/runners/model/FrameworkMethod.java]: @kcooney -- i found an use-case where "isBridge()" is not enough. I think, it should be widened to "isSynthetic()". See https://github.com/junit-team/junit5/compare/ignore_synthetic_methods with a test case included.

Comment (kcooney) [inline src/main/java/org/junit/runners/model/FrameworkMethod.java]: JUnit4 doesn't support using `@Test`, `@Before` etc on anything but "normal" methods.

Comment (sormuras) [inline src/main/java/org/junit/runners/model/FrameworkMethod.java]: ...and JUnit 4 does not provide [support](https://github.com/junit-team/junit5/tree/master/junit-platform-commons/src/main/java/org/junit/platform/commons/support) tools like JUnit 5 now does.

Comment (sormuras) [inline src/main/java/org/junit/runners/model/FrameworkMethod.java]: @kcooney Is the assumption correct, that every bridge method is also synthetic? Couldn't find a definitive answer to that question in the language spec or via googling, yet.

Comment (kcooney) [inline src/main/java/org/junit/runners/model/FrameworkMethod.java]: @sormuras synetetic methods are any methods introduced by the compiler that do not correspond to something in the source (with the exception of default constructors and static initializers). Therefore any bridge method is a synthetic method.

**Pull request merged**